src/meta/greedy_load_balancer.cpp

```diff
@@ -1245,6 +1245,96 @@ bool greedy_load_balancer::pick_up_move(const cluster_migration_info &cluster_in
                                         const int32_t app_id,
                                         const partition_set &selected_pid,
                                         /*out*/ move_info &move_info)
+{
+    rpc_address max_load_node;
+    std::string max_load_disk;
+    partition_set max_load_partitions;
+    get_max_load_disk(
+        cluster_info, max_nodes, app_id, max_load_node, max_load_disk, max_load_partitions);
+
+    for (const auto &node_addr : min_nodes) {
+        gpid picked_pid;
+        if (pick_up_partition(
+                cluster_info, node_addr, max_load_partitions, selected_pid, picked_pid)) {
+            move_info.pid = picked_pid;
+            move_info.source_node = max_load_node;
+            move_info.source_disk_tag = max_load_disk;
+            move_info.target_node = node_addr;
+            move_info.type = cluster_info.type == cluster_balance_type::COPY_SECONDARY
+                                 ? balance_type::copy_secondary
+                                 : balance_type::copy_primary;
+            ddebug_f("partition[{}] will migrate from {} to {}",
+                     picked_pid,
+                     max_load_node.to_string(),
+                     node_addr.to_string());
+            return true;
+        }
+    }
+    return false;
+}
+
+void greedy_load_balancer::get_max_load_disk(const cluster_migration_info &cluster_info,
+                                             const std::set<rpc_address> &max_nodes,
+                                             const int32_t app_id,
+                                             /*out*/ rpc_address &picked_node,
+                                             /*out*/ std::string &picked_disk,
+                                             /*out*/ partition_set &target_partitions)
+{
+    int32_t max_load_size = 0;
+    for (const auto &node_addr : max_nodes) {
+        std::map<std::string, partition_set> disk_partitions =
```

Comment:
I suggest adding comment for this structure: disk_tag -> partition_set

Reply:
good idea